watertap/tools/oli_api/credentials.py

```diff
@@ -191,164 +267,138 @@ def set_access_key(self):
         elif len(self.credentials["access_keys"]) == 1:
             return self.credentials["access_keys"][0]
         else:
-            print(" Specify an access key:")
+            _logger.info("Specify an access key: ")
             for i in range(len(self.credentials["access_keys"])):
-                print(f"{i}\t{self.credentials['access_keys'][i]}")
-            return self.credentials["access_keys"][int(input(" "))]
-
-    # TODO: improve header management for class
-    def generate_oliapi_access_key(self):
+                _logger.info(f"{i}\t{self.credentials['access_keys'][i]}")
+                if self.test:
+                    r = 0
+                else:
+                    r = int(input(" "))
+            return self.credentials["access_keys"][r]
```

Comment:
Can you explain this bit? Not sure what `self.test` is (or rather why it's called `test`), but my guess is this bit of code just allows the user to select the ith entry from a list of access keys.

Reply:
True - `self.test` is there to bypass `input()` lines so that tests are autonomous. It'll be replaced with `interactive_mode`, i.e., where `False` skips all the input lines and shows debug messages as well